src/clj/fluree/db/query/api.cljc

```diff
@@ -209,36 +208,36 @@
         (dataset db-map defaults)))))
 
 (defn query-connection-fql
-  [conn query did]
+  [conn query request-opts]
   (go-try
-   (let [{:keys [t opts] :as query*} (-> query
-                                         syntax/coerce-query
-                                         (update :opts sanitize-query-options did))
-         default-aliases (some-> query* :from util/sequential)
-         named-aliases   (some-> query* :from-named util/sequential)]
-     (if (or (seq default-aliases)
-             (seq named-aliases))
-       (let [s-ctx    (ctx-util/extract query)
-             ds       (<? (load-dataset conn default-aliases named-aliases t
-                                        s-ctx opts))
-             query**  (update query* :opts dissoc :meta :max-fuel ::util/track-fuel?)
-             max-fuel (:max-fuel opts)]
-         (if (::util/track-fuel? opts)
-           (<? (track-query ds max-fuel query**))
-           (<? (fql/query ds query**))))
-       (throw (ex-info "Missing ledger specification in connection query"
-                       {:status 400, :error :db/invalid-query}))))))
-
+    (let [{:keys [t opts] :as sanitized-query} (-> query
+                                                   syntax/coerce-query
+                                                   (update :opts sanitize-query-options (:did request-opts)))
+          
+          default-aliases (some-> sanitized-query :from util/sequential)
+          named-aliases   (some-> sanitized-query :from-named util/sequential)]
+      (if (or (seq default-aliases)
+              (seq named-aliases))
+        (let [s-ctx    (ctx-util/extract query)
+              ds       (<? (load-dataset conn default-aliases named-aliases t
+                                            s-ctx opts))
+              trimmed-query (update sanitized-query :opts dissoc :meta :max-fuel ::util/track-fuel?)
+              max-fuel (:max-fuel opts)]
+          (if (::util/track-fuel? opts)
+            (<? (track-query ds max-fuel trimmed-query))
+            (<? (fql/query ds trimmed-query))))
+        (throw (ex-info "Missing ledger specification in connection query"
+                        {:status 400, :error :db/invalid-query}))))))
 
 (defn query-connection-sparql
-  [conn query did]
+  [conn query {:keys [role did] :as request-opts}]
   (go-try
-   (let [fql (sparql/->fql query)]
-     (<? (query-connection-fql conn fql did)))))
+    (let [fql              (sparql/->fql query)
+          credentialed-fql (update fql :opts #(merge {:role role :did did} %))]
+      (<? (query-connection-fql conn credentialed-fql request-opts)))))
 
 (defn query-connection
-  [conn query {:keys [format did] :as opts :or {format :fql}}]
-  (case format
-    :fql (query-connection-fql conn query did)
-    :sparql (query-connection-sparql conn query did)))
-
+  [conn query {:keys [format] :as request-opts}]
```

Comment:
Instead of `request-opts`, can we just call this `opts`? It's not guaranteed that this will be called during the processing of a request.

Reply:
I was hoping to differentiate it somehow between the opts that we put directly into queries and transactions. Would something like `external-opts` be better?

Reply:
I'm not sure there is a meaningful distinction there, we use opts to refer to both everywhere in the api, so I think it's best to be consistent with that.

Reply:
That's a good point, and I'm a believer that different things should have different names. Can you open a ticket for renaming the opts that aren't part of the document? Then we can decide the best name for it there and get it fixed across the board, rather than just fixing it locally here.